src/CustomElements/v1/README.md

```diff
@@ -0,0 +1,139 @@
+native Shadow DOM v1, and native and polyfilled HTML Imports.
+
+The implementation could use more tests, especially around ordering of
+reactions. The source references old versions of the spec.
+
+### To do
+
+  1. Implement CustomElementsRegistry#whenDefined
+  2. Implement Node#isConnected
+  3. Re-write spec references
+  4. Add reaction callback ordering tests
+  5. Reorganize tests to be closer to spec structure
+  6. Performance tests
+  7. Clarify integration with HTML Imports
+
+## Implementation approach and browser support
+
+The polyfill leans heavily on MutationObservers to drive custom element creation and reactions. This means that the polyfill requires native or polyfilled MutationObservers. The polyfill also uses Map and Set, though those would be easy to polyfill or remove.
+
+The implementation should work without additional polyfills in IE 11, Edge, Safari >6, Firefox, Chrome, Android Browser >4.4, and Opera (not Mini).
+
+IE 10 should be supported with the Mutation Observer polyfill.
+
+This branch does not pass the CI tests yet, but it passes locally in Chrome 52, Safari 9.1, Safari Technical Preview, and Firefox 47.
+
+Because MutationObservers are used to create custom element instances and react to document structure and attribute changes, all reactions are asynchronous while in the specs, some are synchronous. This is by design, since some of the synchronous reactions are impossible to polyfill. We consider it more important that reactions have the same relative timing to each other as the spec, for example attributeChangedCallback happens after connectedCallback.
+
+### Implementing the "Constructor-call Trick"
+
+The HTMLElement constructor is now specified to return a different object than `this`, so that the parser and upgrades can call the constructor on elements that have already been allocated. JavaScript allows this, but some compilers, such as TypeScript and Closure currently don't.
+
+The HTMLElement constructor is also specified to look up the tag name associated with a constructor by using `new.target`. `new.target` isn't available in ES5, is not polyfillable, but `this.constructor` behaves similarly for most ES5 class patterns, including the output of Babel, TypeScript and Closure, so the polyfill uses that to look up tag names.
+
+`new.target` isn't even feature detectable, since it's a syntax error in ES5. Because of this, the polyfill can't check `new.target` first and fallback to `this.constructor`. This also means that ES5-style constructors can't conditionally make a "super" call to the HTMLElement constructor (with `Reflect.construct`) in non-ES6 environments to be compatible with native Custom Elements.
+
+To allow for elements that work in both ES5 and ES6 environments, we provide a shim that overrides the HTMLElement constructor and calls `Reflect.construct` with either the value of `new.target` or `this.constructor`. This shim can only be executed in ES6 environments.
```

Comment:
Oh, and the shim can never be included by default with the polyfill. `new.target` is a syntax error if it's not supported, so you can't feature detect it. If it were included in the polyfill, the whole polyfill would fail to load.